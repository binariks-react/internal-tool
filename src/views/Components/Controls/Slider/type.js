export const SLIDER_TYPE = Object.freeze({
  SINGLE: 'single',
  DOUBLE: 'double',
});

export const CONTROLLER_TYPE = Object.freeze({
  LEFT: 'left-controller',
  RIGHT: 'right-controller',
});
