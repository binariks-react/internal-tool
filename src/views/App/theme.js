export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
