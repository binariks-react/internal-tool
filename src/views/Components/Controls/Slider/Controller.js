import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { CONTROLLER_TYPE } from './type';

const Controller = (props) => {
  const {
    initialValue,
    controllerType,
    handleControllerChange,
    minValue,
    maxValue,
    showValue,
    sliderWidth,
    rangeValue,
  } = props;
  const [controllerPosition, setControllerPosition] = useState(initialValue);
  const [clickedController, setClickedController] = useState(false);
  const [mousePosition, setMousePosition] = useState(NaN);
  const controllerReference = useRef(null);
  const convertedOneValueToPixelPresentation = rangeValue / sliderWidth;
  useEffect(() => {
    const handleMouseDown = (e) => {
      setClickedController(true);
      setMousePosition(e.clientX);
      document.body.style.cursor = 'grabbing';
    };

    const handleTouchStart = (e) => {
      setClickedController(true);
      setMousePosition(e.touches[0].clientX);
    };

    const handleMouseUp = () => {
      if (clickedController) {
        setClickedController(false);
        document.body.style.cursor = 'default';
      }
    };

    const handleTouchEnd = () => {
      if (clickedController) {
        setClickedController(false);
      }
    };

    const controllerMove = (nextPosition) => {
      if (clickedController && mousePosition !== nextPosition) {
        const nextControllerPosition = controllerPosition - (mousePosition - nextPosition);
        if (nextControllerPosition >= minValue && nextControllerPosition <= maxValue) {
          setControllerPosition(nextControllerPosition);
          setMousePosition(nextPosition);
          handleControllerChange(nextControllerPosition, controllerType);
        }
        if (nextControllerPosition <= minValue && controllerPosition !== minValue) {
          setControllerPosition(minValue);
          handleControllerChange(minValue, controllerType);
        }
        if (nextControllerPosition >= maxValue && controllerPosition !== maxValue) {
          setControllerPosition(maxValue);
          handleControllerChange(maxValue, controllerType);
        }
      }
    };

    const handleMouseMove = (e) => {
      controllerMove(e.clientX);
    };

    const handleTouchMove = (e) => {
      controllerMove(Math.ceil(e.touches[0].clientX));
    };

    controllerReference.current.addEventListener('mousedown', handleMouseDown);
    controllerReference.current.addEventListener('touchstart', handleTouchStart);
    document.addEventListener('mousemove', handleMouseMove);
    document.addEventListener('touchmove', handleTouchMove);
    document.addEventListener('mouseup', handleMouseUp);
    document.addEventListener('touchend', handleTouchEnd);

    return () => {
      controllerReference.current.removeEventListener('mousedown', handleMouseDown);
      controllerReference.current.removeEventListener('touchstart', handleTouchStart);
      document.removeEventListener('mousemove', handleMouseMove);
      document.removeEventListener('touchmove', handleTouchMove);
      document.removeEventListener('mouseup', handleMouseUp);
      document.removeEventListener('touchend', handleTouchEnd);
    };
  }, [clickedController, mousePosition]);

  return (
    <ControllerWrapper controllerPosition={controllerPosition} tabindex="0">
      <ControllerButton ref={controllerReference} clickedController={clickedController} tabIndex="0">
        {showValue &&
          <ValueContainer clickedController={clickedController} >
            {(controllerPosition * convertedOneValueToPixelPresentation).toFixed(2)}
          </ValueContainer>
        }
      </ControllerButton>
    </ControllerWrapper>
  );
};

Controller.defaultProps = {
  controllerType: CONTROLLER_TYPE.LEFT,
};

Controller.propTypes = {
  initialValue: PropTypes.number.isRequired,
  controllerType: PropTypes.oneOf([CONTROLLER_TYPE.LEFT, CONTROLLER_TYPE.RIGHT]),
  handleControllerChange: PropTypes.func.isRequired,
  minValue: PropTypes.number.isRequired,
  maxValue: PropTypes.number.isRequired,
  showValue: PropTypes.bool.isRequired,
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
};

export default Controller;

const ControllerWrapper = styled.div.attrs({ 'data-test': 'controller-wrapper' })`
  position: absolute;
  top: 0;
  left: 0;
  transform: translateX(${ props => props.controllerPosition}px);
`;

const ControllerButton = styled.div.attrs({ 'data-test': 'controller-button' })`
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  transition: 250ms;

  ${(props) => {
    if (props.clickedController) {
      return `
          width: 23px;
          height: 23px;
          transform: translate(-17%, -9px); 
          box-shadow: 0 0 15px rgba(0,123,255,.65), 0 3px 15px rgba(90,97,105,.1), 0 2px 3px rgba(90,97,105,.2);
        `;
    }
    return `
        width: 21px;
        height: 21px;
        transform: translate(-15%, -8px);
        box-shadow: 0 3px 15px rgba(90, 97, 105, .1), 0 2px 3px rgba(90, 97, 105, .2);
      `;
  }
}

  &:active {
    cursor: grabbing;
  }

  &:focus {
    outline: none;
    box-shadow: 0 0 7px rgba(0,123,255,.65), 0 3px 15px rgba(90,97,105,.1), 0 2px 3px rgba(90,97,105,.2);
  }
`;

const ValueContainer = styled.div.attrs({ 'data-test': 'controller-text' })`
  position: absolute;
  top: -150%;
  left: 50%;
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff; 
  font-weight: 300;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90, 97, 105, .1), 0 2px 3px rgba(90, 97, 105, .2);
  user-select: none;
  transition: 250ms;

  ${ (props) => {
    if (props.clickedController) {
      return `
        font-size: 15px;
        padding: 7px 11px;
      `;
    }
    return `
      font-size: 14px;
      padding: 5px 10px;
    `;
  }}
`;
