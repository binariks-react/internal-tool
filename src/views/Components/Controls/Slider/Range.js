import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Range = (props) => {
  const { sliderWidth, leftPoint, rightPoint, rangeColour } = props;

  return (
    <RangeWrapper>
      <RangeLine
        leftPoint={leftPoint}
        rightPoint={rightPoint}
        sliderWidth={sliderWidth}
        rangeColour={rangeColour}
      />
    </RangeWrapper>
  );
};

Range.defaultProps = {
  leftPoint: 0,
};

Range.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  leftPoint: PropTypes.number,
  rightPoint: PropTypes.number.isRequired,
  rangeColour: PropTypes.string.isRequired,
};

export default Range;

const RangeWrapper = styled.div`
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90, 97, 105, 0.1);
`;

const RangeLine = styled.div`
  position: absolute;
  top: 0;
  left: ${props => props.leftPoint}px;
  right: ${props => props.sliderWidth - props.rightPoint - 5}px;
  height: 5px;
  border-radius: 5px;
  background-color: ${props => props.theme.colors[props.rangeColour]};
`;
