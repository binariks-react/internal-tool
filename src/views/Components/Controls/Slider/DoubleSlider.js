import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Range from './Range';
import Controller from './Controller';
import { CONTROLLER_TYPE } from './type';

const DoubleSlider = (props) => {
  const {
    sliderWidth,
    rangeValue,
    leftControllerValue,
    rightControllerValue,
    minDistanceBetweenControllers,
    rangeColour,
    showValue,
    getRange,
  } = props;

  const convertedOneValueToPixelPresentation = (rangeValue / sliderWidth);

  const getControllerPosition = controllerValue => Math.ceil(controllerValue / convertedOneValueToPixelPresentation);

  // eslint-disable-next-line max-len
  const getMinDistanceBetweenControllers = () => Math.ceil(minDistanceBetweenControllers / convertedOneValueToPixelPresentation);

  const calculateBorderCollision = (controllerType, positionOfNextController) => {
    const minDistanceBetweenControllersInPixels = getMinDistanceBetweenControllers();
    switch (controllerType) {
    case CONTROLLER_TYPE.LEFT: {
      return positionOfNextController - Math.ceil(minDistanceBetweenControllersInPixels / 2);
    }
    case CONTROLLER_TYPE.RIGHT: {
      return positionOfNextController + Math.ceil(minDistanceBetweenControllersInPixels / 2);
    }
    default: {
      throw new Error('We didn\'t get controller type');
    }
    }
  };

  const [leftController, setLeftController] = useState({
    position: getControllerPosition(leftControllerValue),
    minValue: 0,
    maxValue: calculateBorderCollision(CONTROLLER_TYPE.LEFT, getControllerPosition(rightControllerValue)),
  });
  const [rightController, setRightController] = useState({
    position: getControllerPosition(rightControllerValue),
    minValue: calculateBorderCollision(CONTROLLER_TYPE.RIGHT, getControllerPosition(leftControllerValue)),
    maxValue: sliderWidth,
  });

  const handleControllerChange = (controllerPosition, controllerType) => {
    switch (controllerType) {
    case CONTROLLER_TYPE.LEFT: {
      setLeftController({
        ...leftController,
        position: controllerPosition,
        maxValue: calculateBorderCollision(controllerType, rightController.position),
      });
      setRightController({
        ...rightController,
        minValue: calculateBorderCollision(controllerType, controllerPosition),
      });
      break;
    }
    case CONTROLLER_TYPE.RIGHT: {
      setLeftController({
        ...leftController,
        maxValue: calculateBorderCollision(controllerType, controllerPosition),
      });
      setRightController({
        ...rightController,
        position: controllerPosition,
        minValue: calculateBorderCollision(controllerType, leftController.position),
      });
      break;
    }
    default: {
      return new Error('Problem with controllers');
    }
    }
    getRange({
      leftPoint: 0,
      rightPoint: (controllerPosition * convertedOneValueToPixelPresentation).toFixed(2),
    });
  };

  return (
    <SliderWrapper sliderWidth={sliderWidth} showValue={showValue} >
      <Range
        id="tickmarks"
        sliderWidth={sliderWidth}
        leftPoint={leftController.position}
        rightPoint={rightController.position}
        rangeColour={rangeColour}
      />
      <Controller
        initialValue={leftController.position}
        controllerType={CONTROLLER_TYPE.LEFT}
        handleControllerChange={handleControllerChange}
        minValue={leftController.minValue}
        maxValue={leftController.maxValue}
        showValue={showValue}
        sliderWidth={sliderWidth}
        rangeValue={rangeValue}
      />
      <Controller
        initialValue={rightController.position}
        controllerType={CONTROLLER_TYPE.RIGHT}
        handleControllerChange={handleControllerChange}
        minValue={rightController.minValue}
        maxValue={rightController.maxValue}
        showValue={showValue}
        sliderWidth={sliderWidth}
        rangeValue={rangeValue}
      />

    </SliderWrapper>
  );
};

export default DoubleSlider;

DoubleSlider.defaultProps = {};

DoubleSlider.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
  leftControllerValue: PropTypes.number.isRequired,
  rightControllerValue: PropTypes.number.isRequired,
  minDistanceBetweenControllers: PropTypes.number.isRequired,
  rangeColour: PropTypes.string.isRequired,
  getRange: PropTypes.func.isRequired,
  showValue: PropTypes.bool.isRequired,
};

const SliderWrapper = styled.div`
  position: relative;
  margin: ${ props => props.showValue ? '55px 0 35px' : '35px 0'};
  width: ${ props => props.sliderWidth}px;
`;
