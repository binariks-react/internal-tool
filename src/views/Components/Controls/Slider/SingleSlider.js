import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Range from './Range';
import Controller from './Controller';

const SingleSlider = (props) => {
  const { sliderWidth, rangeValue, controllerValue, rangeColour, showValue, getRange } = props;
  const convertedOneValueToPixelPresentation = (rangeValue / sliderWidth);

  // eslint-disable-next-line max-len
  const getControllerPosition = controllerValueForCalculating => Math.ceil(controllerValueForCalculating / convertedOneValueToPixelPresentation);

  const [controllerPosition, setControllerPosition] = useState(getControllerPosition(controllerValue));

  const handleControllerChange = (newControllerPosition) => {
    setControllerPosition(newControllerPosition);

    getRange({
      leftPoint: 0,
      rightPoint: (newControllerPosition * convertedOneValueToPixelPresentation).toFixed(2),
    });
  };

  return (
    <SliderWrapper data-test="single-slider" sliderWidth={sliderWidth} showValue={showValue} >
      <Range
        sliderWidth={sliderWidth}
        leftPoint={0}
        rightPoint={controllerPosition}
        rangeColour={rangeColour}
      />
      <Controller
        initialValue={controllerPosition}
        handleControllerChange={handleControllerChange}
        minValue={0}
        maxValue={sliderWidth}
        showValue={showValue}
        sliderWidth={sliderWidth}
        rangeValue={rangeValue}
      />
    </SliderWrapper>
  );
};

SingleSlider.defaultProps = {
  controllerValue: 0,
};

SingleSlider.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
  controllerValue: PropTypes.number,
  rangeColour: PropTypes.string.isRequired,
  showValue: PropTypes.bool.isRequired,
  getRange: PropTypes.func.isRequired,
};

export default SingleSlider;

const SliderWrapper = styled.div.attrs({ 'data-test': 'slider-wrapper' })`
  position: relative;
  margin: ${ props => props.showValue ? '55px 0 35px' : '35px 0'};
  width: ${ props => props.sliderWidth}px;
`;
