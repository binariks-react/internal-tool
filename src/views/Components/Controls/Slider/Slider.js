import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import SingleSlider from './SingleSlider';
import DoubleSlider from './DoubleSlider';

import { SLIDER_TYPE } from './type';

const Slider = props => (
  <React.Fragment>
    {props.type === SLIDER_TYPE.SINGLE &&
      <SingleSlider
        {...props}
        controllerValue={
          props.controllerValue >= 0 && props.controllerValue <= props.rangeValue ?
            props.controllerValue : props.controllerValue > props.rangeValue ? props.rangeValue : 0
        }
      />
    }
    {props.type === SLIDER_TYPE.DOUBLE &&
      <DoubleSlider
        {...props}
        leftControllerValue={props.leftControllerValue >= 0 ? props.leftControllerValue : 0}
        rightControllerValue={props.rightControllerValue <= props.rangeValue ?
          props.rightControllerValue :
          props.rangeValue}
      />
    }
    {
      props.type !== SLIDER_TYPE.SINGLE &&
      props.type !== SLIDER_TYPE.DOUBLE &&
      (<ErrorMessage>You didn't pass normal type of range</ErrorMessage>)
    }
  </React.Fragment>
);

Slider.defaultProps = {
  rangeColour: 'primary',
  showValue: false,
};

Slider.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
  type: PropTypes.oneOf([SLIDER_TYPE.SINGLE, SLIDER_TYPE.DOUBLE]).isRequired,
  // eslint-disable-next-line react/require-default-props
  controllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  leftControllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  rightControllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  minDistanceBetweenControllers: PropTypes.number,
  rangeColour: PropTypes.string,
  getRange: PropTypes.func.isRequired,
  showValue: PropTypes.bool,
};

export default Slider;

const ErrorMessage = styled.div``;
