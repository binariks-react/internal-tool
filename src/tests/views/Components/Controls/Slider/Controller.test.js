import React from 'react';
import { render, mount } from 'enzyme';
import theme from 'views/App/theme';
import 'jest-styled-components';
import 'react-test-renderer/shallow';

import Controller from 'views/Components/Controls/Slider/Controller';
import { CONTROLLER_TYPE } from 'views/Components/Controls/Slider/type';
import { getByTestId } from './testUtils';

describe('<Controller />', () => {
  const fakeProps = {
    initialValue: 0,
    controllerType: CONTROLLER_TYPE.LEFT,
    handleControllerChange: () => { },
    minValue: 0,
    maxValue: 500,
    showValue: false,
    sliderWidth: 500,
    rangeValue: 500,
    theme,
  };

  describe('Snapshot', () => {
    it('Controller', () => {
      const controller = render(
        <Controller
          initialValue={0}
          controllerType={CONTROLLER_TYPE.LEFT}
          handleControllerChange={() => { }}
          minValue={0}
          maxValue={300}
          showValue={false}
          sliderWidth={500}
          rangeValue={1000}
        />
      );
      expect(controller).toMatchSnapshot();
    });
  });

  it('Should contain Button controller', () => {
    const wrapper = render(<Controller {...fakeProps} />);

    const buttonControllerNode = getByTestId(wrapper, 'controller-button');
    expect(buttonControllerNode.length).toBe(1);
  });

  it('Not showing value when we pass showValue: false', () => {
    const wrapper = render(<Controller {...fakeProps} />);

    const textValueNode = getByTestId(wrapper, 'controller-text');
    expect(textValueNode.length).toBe(0);
  });

  it('Show value when pased showValue: true', () => {
    const wrapper = render(<Controller {...fakeProps} showValue />);
    const textValueNode = getByTestId(wrapper, 'controller-text');
    expect(textValueNode.length).toBe(1);
  });

  it('Contain value', () => {
    const wrapper = mount(<Controller {...fakeProps} initialValue={0} showValue />);
    expect(getByTestId(wrapper, 'controller-text').text()).toBe('0.00');

    wrapper.unmount();
    const wrapperNew = mount(<Controller {...fakeProps} initialValue={70} showValue />);

    expect(getByTestId(wrapperNew, 'controller-text').text()).toBe('70.00');
  });
});
