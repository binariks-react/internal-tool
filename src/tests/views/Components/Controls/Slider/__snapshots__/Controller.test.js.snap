// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controller /> Snapshot Controller 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c1:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

<div
  class="c0"
  data-test="controller-wrapper"
>
  <div
    class="c1"
    data-test="controller-button"
    tabindex="0"
  />
</div>
`;
